keps/sig-apps/5440-mutable-job-pod-resource-updates/README.md

```diff
@@ -0,0 +1,583 @@
+
+#### Story 3
+
+As a cluster administrator, I want to monitor existing workloads and see how many resources they are actually using. If a workload is oversubscribed and their actual utilization is lower, I want to suspend that workload and checkpoint the workload via external APIs. I would then lower the request requirements to match the actual utilization and resubmit my job.
+
+### Risks and Mitigations
+
+- New API calls from queue controllers to update resource specifications. The mitigation
+  is for such controllers to make a single API call for both updating resources and
+  unsuspending the job.
+  
+- Potential for resource specification changes to make a job unschedulable if the
+  updated requirements exceed available cluster capacity. Queue controllers should
+  validate resource availability before making changes.
+
+- A race condition could theoretically happen if a job is unsuspended and then quickly
+  suspended again before resource updates, though this is not a typical use case pattern.
+
+## Design Details
+
+The pod template validation logic in the API server needs to be updated to relax the validation
+of the Job's Template field. Currently the template is immutable, but we need to make
+container resource specifications (CPU, memory, GPU, and extended resources requests and limits) mutable for suspended jobs.
+
+The condition we will check to verify that the job is suspended is `Job.Spec.Suspend=true`.
+
+We will allow updates to the following fields in container specifications within the pod template:
+- `resources.requests.cpu`
+- `resources.requests.memory`
+- `resources.requests.*` (for extended resources like `nvidia.com/gpu`, `amd.com/gpu`, `tpu-v4` etc.)
+- `resources.limits.cpu`
+- `resources.limits.memory`
+- `resources.limits.*` (for extended resources like `nvidia.com/gpu`, `amd.com/gpu`, `tpu-v4` etc.)
+
+### DRA Support
+
+DRA does not allow changing ResourceClaimTemplates once they are created.
+At the moment, relaxing mutability constraints of ResourceClaimTemplates or ResourceClaims is not in scope.
+To add support for this feature with DRA, the recommendation is to recreate ResourceClaimTemplates that match the
+desired resources.
+
+One does not have to modify claims in the PodTemplate so one can still assume claims are immutable also.
+
+### Resuming on running workloads
+
+When mutable pod scheduling requirements KEP was implemented, Jobs can only have their scheduling contraints changed on Job that is created in a suspended state.
+
+If a workload is resumed, then the workload is assumed to be immutable from then on. So users are not able to change scheduling constraints or update resources after a Job starts.
+
+In this work, we want to relax this solution to enable #story-3. Users would be able to suspend a running workload, and change the resources on the suspended job.
```

Comment:
When a job is suspended, its running pods are terminated. Is that the expected behavior here as well? It would be helpful to explicitly state that in this section, as it's a critical detail for anyone implementing "Story 3"

Reply:
```suggestion
In this work, we want to relax this solution to enable #story-3. Users would be able to suspend a running workload, and change the resources on the suspended job. It is important to note that when a running Job is suspended, any of its active Pods will be terminated. This is a critical detail for any user or controller implementing this workflow
```